### PR TITLE
Add new string assertions: `BeParseableInto<T>` and `NotBeParseableInto<T>`

### PR DESCRIPTION
**Pull request by ITaluone**
Closes: #168 

## IMPORTANT 

* [x] If the PR touches the public API, the changes have been approved in a separate issue with the "api-approved" label.
* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/main/Tests/AwesomeAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [x] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/main/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://awesomeassertions.org/releases).
* [x] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/main/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/main/AcceptApiChanges.sh).
* [ ] If the PR affects [the documentation](../tree/main/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://awesomeassertions.org/introduction).
    * [ ] Please also run `./build.sh --target spellcheck` or `.\build.ps1 --target spellcheck` before pushing and check the good outcome

### CODE REVIEW

File: Src/AwesomeAssertions/Primitives/StringAssertions.cs [hunk 1]
```diff
@@ -234,6 +235,126 @@ public AndConstraint<TAssertions> NotBeEquivalentTo(string unexpected,
...
+            exception = ex;
+        }
+
+        CurrentAssertionChain
+            .WithExpectation("Expected {context:the subject} with value {0} to be parseable into {1}{reason}, ",
+                Subject, typeof(T),
+                chain => chain
+                    .BecauseOf(because, becauseArgs)
+                    .ForCondition(canBeParsed)
+                    .FailWith($"but it could not. Exception message: {exception?.Message}"));
```
Comment (ITaluone) [inline Src/AwesomeAssertions/Primitives/StringAssertions.cs]: 🤔 Should we make a compound "because" here? With user provided because string + parse exception message.

This would read something like this:
`Expected sut with value "12.34" to be parseable into decimal, because this is a decimal and because the input string '12.34' was not in a correct format, but it could not.`

or like this

`Expected sut with value "12.34" to be parseable into decimal, because this is a decimal, but it could not,  because the input string '12.34' was not in a correct format.`

File: Src/AwesomeAssertions/Primitives/StringAssertions.cs [hunk 2]
```diff
@@ -234,6 +235,126 @@ public AndConstraint<TAssertions> NotBeEquivalentTo(string unexpected,
...
+        params object[] becauseArgs)
+        where T : IParsable<T>
+    {
+        Guard.ThrowIfArgumentIsNull(formatProvider);
+
+        bool canBeParsed = false;
+
+        try
+        {
+            _ = T.Parse(Subject, formatProvider);
```
Comment (ITaluone) [inline Src/AwesomeAssertions/Primitives/StringAssertions.cs]: Are we interested in the exception here? Because in not I will rewrite this to simply `TryParse()`?

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        {
+            // Arrange
+            string sut = "12.34";
+
+            // Act
+            Action act = () => sut.Should().BeParsableInto<decimal>(new CultureInfo("de-AT"));
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage("*12.34*decimal*could not*not in a correct format*");
+        }
```
Comment (ITaluone) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: This test fails on macOS, but succeeds on windows/linux


<details><summary>ChatGPT's suggestion why</summary>
<p>
❌ Actual behavior on macOS:
On macOS, parsing "12.34" as a decimal with de-AT succeeds, when it shouldn't.

🧠 Why this happens: ICU vs Windows Globalization
.NET Core (and .NET 5+) on non-Windows systems use the ICU (International Components for Unicode) library for globalization by default.

However, macOS ships its own version of ICU — which can behave differently than Linux/Windows, especially for cultures like de-AT.

In particular:
On macOS, some cultures (like de-AT) have locale fallbacks or different parsing leniencies in ICU.

It seems that ICU on macOS accepts both . and , as decimal separators in certain locales — a behavior that doesn’t happen on Linux (even though both use ICU).

✅ Fix/Workaround
There are a few ways to ensure consistent behavior across platforms:

1. Force invariant globalization
You can run your tests with .NET's invariant globalization mode, which removes platform variability:

Set environment variable:
```bash
DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=true
```

But this disables culture-specific parsing altogether — not what you want here.

2. Use NumberStyles to force strict parsing
Instead of using FluentAssertions.BeParsableInto<T>() (which may rely on loose parsing), try writing a stricter test:

```csharp
CultureInfo culture = new CultureInfo("de-AT");
string sut = "12.34";

Action act = () => Convert.ToDecimal(sut, culture);

act.Should().Throw<FormatException>();
```

This avoids relying on FluentAssertions' behavior and tests the actual parsing edge case.

3. Compare separators explicitly in the test
```csharp
Assert.Equal(',', new CultureInfo("de-AT").NumberFormat.NumberDecimalSeparator[0]);
```

Then test parsing with the wrong separator and expect failure.

✅ Final Recommendation
Replace the test with an explicit decimal.TryParse or Convert.ToDecimal call using the desired culture and NumberStyles.Number, and assert that it throws — avoiding FluentAssertions' abstraction if platform behavior is inconsistent.
</p>
</details> 

What would you go for here @cbersch?

Comment (ITaluone) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: Current approach works, but could give some strange results when used on macOS.

Or should we take every peculiarity of `IParsable.Parse()`, like this one, because that's how it works on macOS?

File: Src/AwesomeAssertions/Primitives/StringAssertions.cs (hunk 1 above)
Comment (cbersch) [reply Src/AwesomeAssertions/Primitives/StringAssertions.cs]: ~~I thought, that the exception message might give more information, but it doesn't help in our case.~~

I prefer the second variant.

File: Src/AwesomeAssertions/Primitives/StringAssertions.cs (hunk 2 above)
Comment (cbersch) [reply Src/AwesomeAssertions/Primitives/StringAssertions.cs]: ~~Seems like the exception doesn't provide any benefit.~~
In case of `Guid` the exception message provides more information.

File: Src/AwesomeAssertions/Primitives/StringAssertions.cs [hunk 4]
```diff
@@ -234,6 +235,126 @@ public AndConstraint<TAssertions> NotBeEquivalentTo(string unexpected,
         return new AndConstraint<TAssertions>((TAssertions)this);
     }
 
+#if NET8_0_OR_GREATER
+
+    /// <summary>
+    /// Asserts that a string is parseable into something else, which is implementing <see cref="IParsable{TSelf}"/>.
```
Comment (cbersch) [inline Src/AwesomeAssertions/Primitives/StringAssertions.cs]: parsable (there are more typos of this kind)

File: Src/AwesomeAssertions/Primitives/StringAssertions.cs [hunk 5]
```diff
@@ -234,6 +235,126 @@ public AndConstraint<TAssertions> NotBeEquivalentTo(string unexpected,
...
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    /// <typeparam name="T">The type to what the <see cref="string"/> should parsed to.</typeparam>
+    public AndWhichConstraint<TAssertions, T> BeParsableInto<T>(string because = "", params object[] becauseArgs)
+        where T : IParsable<T>
+    {
+        return BeParsableInto<T>(CultureInfo.InvariantCulture, because, becauseArgs);
```
Comment (cbersch) [inline Src/AwesomeAssertions/Primitives/StringAssertions.cs]: I would pass `null` to let the type choose its default`

File: Src/AwesomeAssertions/Primitives/StringAssertions.cs [hunk 6]
```diff
@@ -234,6 +235,126 @@ public AndConstraint<TAssertions> NotBeEquivalentTo(string unexpected,
...
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    /// <typeparam name="T">The type to what the <see cref="string"/> should parsed to.</typeparam>
+    public AndWhichConstraint<TAssertions, T> BeParsableInto<T>(IFormatProvider formatProvider, string because = "",
+        params object[] becauseArgs)
+        where T : IParsable<T>
+    {
+        Guard.ThrowIfArgumentIsNull(formatProvider);
```
Comment (cbersch) [inline Src/AwesomeAssertions/Primitives/StringAssertions.cs]: `null` is a valid value

File: Src/AwesomeAssertions/Primitives/StringAssertions.cs [hunk 7]
```diff
@@ -234,6 +235,126 @@ public AndConstraint<TAssertions> NotBeEquivalentTo(string unexpected,
...
+
+        T parsed = default;
+        var canBeParsed = true;
+        Exception exception = null;
+
+        try
+        {
+            parsed = T.Parse(Subject, formatProvider);
+        }
+        catch (Exception ex) when (ex is OverflowException or FormatException)
```
Comment (cbersch) [inline Src/AwesomeAssertions/Primitives/StringAssertions.cs]: We should catch all exceptions and save only their message.

File: Src/AwesomeAssertions/Primitives/StringAssertions.cs [hunk 8]
```diff
@@ -234,6 +235,126 @@ public AndConstraint<TAssertions> NotBeEquivalentTo(string unexpected,
...
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndConstraint<TAssertions> NotBeParsableInto<T>(string because = "", params object[] becauseArgs)
+        where T : IParsable<T>
+    {
+        return NotBeParsableInto<T>(CultureInfo.InvariantCulture, because, becauseArgs);
```
Comment (cbersch) [inline Src/AwesomeAssertions/Primitives/StringAssertions.cs]: `(null, because, becauseArgs)`

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage("*95e11*int*we want*failure message*could not*");
+        }
+
+        [Theory]
+        [InlineData("de-AT")]
+        [InlineData("en-US")]
+        [InlineData("zh-CN")]
+        [InlineData("ru-RU")]
+        public void A_guid_string_is_parsable_into_a_guid_in_various_cultures(string cultureName)
```
Comment (cbersch) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: I would change this test to check culture-specific parsing of floats. I don't think, that culture-specific guid parsing is relevant for us, but only, that the format provider is passed as expected.

```C#
[InlineData("de-AT", "12,34", 12.34)]
[InlineData("en-US", "12.34", 12.34)]
```

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+        {
+            // Arrange
+            var sut = "12,34";
+
+            // Act / Assert
+            sut.Should().BeParsableInto<decimal>(new CultureInfo("de-AT"))
+                .Which.Should().Be(12.34M);
+        }
+
+        // This test mails on macOS, because of how macOS handles decimal parsing.
```
Comment (ITaluone) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]:
```suggestion
        // This test fails on macOS, because of how macOS handles decimal parsing.
```

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+        {
+            // Arrange
+            string sut = "95e117d5-bb0d-4b16-a1c7-a11eea0284af";
+
+            // Act / Assert
+            sut.Should().NotBeParsableInto<int>();
+        }
+
+        [Fact]
+        public void A_guid_string_which_is_parsable_into_a_guid_but_is_expected_to_not_be_throws()
```
Comment (cbersch) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: This test doesn't give any benefit over the next. I think it's enough to check this once, either int or Guid, but not both.

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+
+            // Assert
+            act.Should().Throw<ArgumentNullException>();
+        }
+    }
+
+    public class NotBeParseableInto
+    {
+        [Fact]
+        public void An_invalid_guid_string_is_parsable_into_a_guid()
```
Comment (cbersch) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: is_not_parsable

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 13]
```diff
@@ -0,0 +1,230 @@
...
+
+            // Act
+            Action act = () => sut.Should().BeParsableInto<byte>(new CultureInfo("de-AT"));
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage("*1234*byte*too large or too small*byte*");
+        }
+
+        [Fact]
+        public void A_null_string_throws()
```
Comment (cbersch) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: Please remove, because this test doesn't check our code, but the `Guid.Parse`. And it is up to the `IParsable` implementation to decide if `null` can be parsed.

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs (hunk 3 above)
Comment (cbersch) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: We take the `IParsable.Parse` as it is on the respective platform. That's not up to us to decide. We only invoke the interface.

That's also why I would strip down the tests. We don't need to check different types, just those which show, e.g. that the culture was passed correctly.

Have you checked with "de-DE" or parsing "12,34" with "en-US"?

**Review by cbersch — CHANGES_REQUESTED** [review]

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs (hunk 13 above)
Comment (IT-VBFK) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: In FA these tests were "required", so I added them here

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs (hunk 3 above)
Comment (IT-VBFK) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: Ok, then I leave it with the optional compiling.. because otherwise we have failing tests on macOS

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs (hunk 13 above)
Comment (cbersch) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: They are, if they are relevant, i.e. if we check for `null`, which we must not do for the string to parse: `MyParsable.Parse(null)` can be perfectly valid.

Comment (IT-VBFK) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: Maybe I add a custom class which implements `IParsable<T>` to cover that as well?

Comment (cbersch) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: I had also thought about this, but I find it unnecessary, because we only provide a wrapper for the type under test, and must not test its implementation details.

Comment (IT-VBFK) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: Ok, I leave it there for now.. we can drop that commit anytime later :)

File: Src/AwesomeAssertions/Primitives/StringAssertions.cs [hunk 14]
```diff
@@ -234,6 +236,125 @@ public AndConstraint<TAssertions> NotBeEquivalentTo(string unexpected,
...
+    public AndConstraint<TAssertions> NotBeParsableInto<T>(IFormatProvider formatProvider, string because = "",
+        params object[] becauseArgs)
+        where T : IParsable<T>
+    {
+        bool canBeParsed = false;
+        string exceptionMessage = null;
+
+        try
+        {
+            _ = T.Parse(Subject, formatProvider);
```
Comment (cbersch) [inline Src/AwesomeAssertions/Primitives/StringAssertions.cs]: Here I would use `TryParse`, because the assertion should fail for successful parsing, in which case we don't have an exception for the error message.

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 15]
```diff
@@ -0,0 +1,279 @@
...
+
+            // Act
+            Action act = () => sut.Should().BeParsableInto<int>("we want to test the {0}", "failure message");
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage("*95e11*int*we want*failure message*could not*");
+        }
+
+        [Fact]
+        public void Chaining_will_be_ignore_when_parsing_fails()
```
Comment (cbersch) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: Typo: ignored

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 16]
```diff
@@ -0,0 +1,279 @@
...
+            // Act
+            Action act = () =>
+                sut.Should().BeParsableInto<int>(new CultureInfo("de-AT"), "we want to test the {0}", "failure message");
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage("*95e11*int*we want*failure message*could not*");
+        }
+
+        [Fact]
+        public void Can_parse_a_floating_point_number_with_correct_culture_specific_separator()
```
Comment (cbersch) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: This test duplicates `A_floating_point_string_is_parsable_into_a_floating_point_value_in_various_cultures`

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 17]
```diff
@@ -0,0 +1,279 @@
...
+            // Arrange
+            var sut = "12,34";
+
+            // Act / Assert
+            sut.Should().BeParsableInto<decimal>(new CultureInfo("de-AT"))
+                .Which.Should().Be(12.34M);
+        }
+
+        [Fact]
+        public void Parsing_with_culture_specific_format_works()
```
Comment (cbersch) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: This test should fail as it is now.

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 18]
```diff
@@ -0,0 +1,279 @@
...
+        public void Parsing_into_a_custom_type_works()
+        {
+            string sut = "blah";
+
+            sut.Should().BeParsableInto<BlahParsableType>().Which
+                .Value.Should().Be("(BlahParsableType): blah");
+        }
+
+        [Fact]
+        public void Parsing_invalid_data_into_a_custom_type_throws()
```
Comment (cbersch) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: In my opinion we need much less tests. The following four tests don't lack anything:

1. BeParsableInto_ValidString_ReturnsExpectedParsedValue
2. BeParsableInto_ValidStringForSpecificCulture_ReturnsExpectedParsedValue
3. BeParsableInto_InvalidString_ThrowsWithProperMessage
4. BeParsableInto_InvalidStringForWrongFormatProvider_ThrowsWithProperMessage

And four respective tests for `NotBeParsableInto`.

I cannot type the full tests, because I only have access to a phone for the next days.

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 19]
```diff
@@ -0,0 +1,279 @@
...
+        {
+            // Arrange
+            string sut = "95e117d5bb0d4b16a1c7a11eea0284a";
+
+            // Act / Assert
+            sut.Should().NotBeParsableInto<Guid>();
+        }
+
+        [Fact]
+        public void A_guid_string_is_not_parsable_into_an_int()
```
Comment (cbersch) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: This test doesn't add any additional value to `An_invalid_guid_string_is_not_parsable_into_a_guid`

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs (hunk 17 above)
Comment (IT-VBFK) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: Why? This is what we get from `Parse`. And we settled that we rely on what `Parse` gives us

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs (hunk 18 above)
Comment (IT-VBFK) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: I would add tow tests each:
- a valid parse + chaining which
- an invalid parse + (ignored) chaining which

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs (hunk 17 above)
Comment (IT-VBFK) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: Ahh.. Chaining `Which` revealed what's going on: it is _parsable_, but it is parsed into 1234 instead of 12.34

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs (hunk 18 above)
Comment (IT-VBFK) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: FYI: the `Cannot_parse_a_larger_integer_to_a_type_which_can_hold_less_places` was meant for testing every possible exception `Parse` can give us

Comment (cbersch) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: For me it would be enough to only have valid parsing plus chaining. Having one test only for valid parsing, without the chaining adds no benefit

Comment (IT-VBFK) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: Hmm.. seems correct, since how assertion chain is designed, it won't get further down than the first fail

Comment (cbersch) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: > FYI: the `Cannot_parse_a_larger_integer_to_a_type_which_can_hold_less_places` was meant for testing every possible exception `Parse` can give us

We must catch all exceptions, because we must also handle, whatever custom implementations throw.
Therefore, it is enough to have one test why throws any exception.

Comment (IT-VBFK) [reply Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: Yeah I see :)

with the `catch (Exception ex)` with no restriction it makes no sense aynmore

**Review by ScarletKuro — APPROVED** [review]

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+        }
+
+        [Fact]
+        public void An_integer_string_is_parsable_into_an_int_but_throws_when_not_expected_to_with_culture_defined()
+        {
+            // Arrange
+            string sut = "1";
+
+            // Act
+            Action act = () => sut.Should().NotBeParsableInto<int>(new CultureInfo("de-DE"),"we want to test the {0}", "failure message");
```
Comment (IT-VBFK) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]: I think one test is enough to test if `because` / `becauseArgs` are passed down the line..

So:
```suggestion
            Action act = () => sut.Should().NotBeParsableInto<int>(new CultureInfo("de-DE"));
```

But in return we should test, if the exception message is correct (especially the casing).

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 21]
```diff
@@ -0,0 +1,121 @@
...
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage("*1*int*we want*failure message*could*");
```
Comment (IT-VBFK) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]:
```suggestion
            act.Should().Throw<XunitException>().WithMessage("*1*int*but it could*");
```

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 22]
```diff
@@ -0,0 +1,121 @@
...
+
+        [Fact]
+        public void A_string_is_not_parsable_into_an_int_even_with_a_culture_defined()
+        {
+            // Arrange
+            string sut = "aaa";
+
+            // Act
+            Action act = () =>
+                sut.Should().BeParsableInto<int>(new CultureInfo("de-AT"), "we want to test the {0}", "failure message");
```
Comment (IT-VBFK) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]:
```suggestion
                sut.Should().BeParsableInto<int>(new CultureInfo("de-AT"));
```

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 23]
```diff
@@ -0,0 +1,121 @@
...
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage("*aaa*int*we want*failure message*could not*");
```
Comment (IT-VBFK) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]:
```suggestion
            act.Should().Throw<XunitException>().WithMessage("*aaa*int*the string*was not in the correct format*");
```

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 24]
```diff
@@ -0,0 +1,121 @@
...
+        {
+            // Arrange
+            string sut = "aaa";
+
+            // Act
+            Action act = () =>
+                sut.Should().BeParsableInto<int>(new CultureInfo("de-AT"));
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage("*aaa*int*the string*was not in the correct format*");
```
Comment (IT-VBFK) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]:
```suggestion
            act.Should().Throw<XunitException>().WithMessage("*aaa*int*the input string*was not in the correct format*");
```

File: Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs [hunk 25]
```diff
@@ -0,0 +1,121 @@
...
+        {
+            // Arrange
+            string sut = "aaa";
+
+            // Act
+            Action act = () =>
+                sut.Should().BeParsableInto<int>(new CultureInfo("de-AT"));
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage("*aaa*int*the input string*was not in the correct format*");
```
Comment (IT-VBFK) [inline Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Parsable.cs]:
```suggestion
            act.Should().Throw<XunitException>().WithMessage("*aaa*int*the input string*was not in a correct format*");
```

**Review by IT-VBFK — APPROVED** [review]
